src/static/static_context.rs

```diff
@@ -22,13 +22,24 @@ impl FutharkContext {
         }
     }
 
-    pub(crate) fn ptr(&mut self) -> *mut bindings::futhark_context {
-        self.context
+    pub(crate) fn ptr(&self) -> *mut bindings::futhark_context {
+        unsafe {
+            std::mem::transmute(self.context)
```

Comment:
This looks risky at best, likely unsound to me. 

```rust
fn ptr_mut(&mut self) -> *mut bindings::futhark_context {}
fn ptr(&self) -> *const bindings::futhark_context {}
```

would be the correct way